src/libs/migrations/PersonalDetailsByAccountID.js

```diff
@@ -0,0 +1,213 @@
+    });
+}
+
+/**
+ * We use the old personalDetails object becuase it is more efficient for this migration since it is keyed by email address.
+ * Also, if the old personalDetails object is not available, that likely means the migration has already run successfully before on this account.
+ *
+ * @returns {Promise<Object>}
+ */
+function getDeprecatedPersonalDetailsFromOnyx() {
+    return new Promise((resolve) => {
+        const connectionID = Onyx.connect({
+            key: DEPRECATED_ONYX_KEYS.PERSONAL_DETAILS,
+            callback: (allPersonalDetails) => {
+                Onyx.disconnect(connectionID);
+                return resolve(allPersonalDetails);
+            },
+        });
+    });
+}
+
+/**
+ * Migrate Onyx data for the email to accountID migration.
+ *
+ * - reportAction_
+ *     - originalMessage.oldLogin -> originalMessage.oldAccountID
+ *     - originalMessage.newLogin -> originalMessage.newAccountID
+ *     - accountEmail -> accountID
+ *     - actorEmail -> actorAccountID
+ *     - childManagerEmail -> childManagerAccountID
+ *     - whisperedTo -> whisperedToAccountIDs
+ *     - childOldestFourEmails -> childOldestFourAccountIDs
+ *     - originalMessage.participants -> originalMessage.participantAccountIDs
+ *
+ * @returns {Promise<void>}
+ */
+export default function () {
+    return Promise.all([getReportActionsFromOnyx(), getDeprecatedPersonalDetailsFromOnyx()]).then(([oldReportActions, oldPersonalDetails]) => {
+        const onyxData = {};
+
+        if (!oldReportActions) {
+            Log.info('[Migrate Onyx] Skipped migration PersonalDetailsByAccountID because there were no reportActions');
+            return;
+        }
+
+        // We migrate reportActions to have the new accountID-based data if they don't already.
+        // If we are not able to get the accountID from personalDetails for some reason, we will just clear the reportAction
+        // and let it be fetched from the API next time they open the report and scroll to that action.
+        // We do this because we know the reportAction from the API will include the needed accountID data.
+        _.each(oldReportActions, (reportActionsForReport, onyxKey) => {
+            if (_.isEmpty(reportActionsForReport)) {
+                Log.info(`[Migrate Onyx] Skipped migration PersonalDetailsByAccountID for ${onyxKey} because there were no reportActions`);
+                return;
+            }
+
+            const newReportActionsForReport = {};
+            let reportActionsWereModified = false;
+            _.each(reportActionsForReport, (reportAction, reportActionID) => {
+                if (_.isEmpty(reportAction)) {
+                    reportActionsWereModified = true;
+                    Log.info(`[Migrate Onyx] PersonalDetailsByAccountID migration: removing reportAction ${reportActionID} because the reportAction was empty`);
+                    return;
+                }
+
+                const newReportAction = reportAction;
```

Comment:
Why we are adding a new variable here? Both `newReportAction` and `reportAction` refer to the same object.

Reply:
Hmm but under the hood we are doing exactly that right? if you console.log reportAction it will be exactly the same as newReportAction

Reply:
I agree. Do you see it as a problem though? Is there a reason for me to copy the object instead?

Reply:
No, the objects here are used only once. All good :+1:

Reply:
Thanks!